numpy/core/src/multiarray/descriptor.c

```diff
@@ -1099,34 +1085,34 @@ _convert_from_dict(PyObject *obj, int align)
             }
             PyErr_Clear();
         }
-        tup = PyTuple_New(len);
-        descr = PyObject_GetItem(descrs, ind);
+        PyObject *tup = PyTuple_New(len);
+        PyObject *descr = PyObject_GetItem(descrs, ind);
         if (!descr) {
             Py_DECREF(tup);
             Py_DECREF(ind);
             goto fail;
         }
-        newdescr = _arraydescr_run_converter(descr, align);
+        PyArray_Descr *newdescr = _arraydescr_run_converter(descr, align);
         Py_DECREF(descr);
         if (newdescr == NULL) {
             Py_DECREF(tup);
             Py_DECREF(ind);
             goto fail;
         }
         PyTuple_SET_ITEM(tup, 0, (PyObject *)newdescr);
+        int _align = 1;
         if (align) {
             _align = newdescr->alignment;
             maxalign = PyArray_MAX(maxalign,_align);
         }
         if (offsets) {
-            long offset;
-            off = PyObject_GetItem(offsets, ind);
+            PyObject *off = PyObject_GetItem(offsets, ind);
             if (!off) {
                 Py_DECREF(tup);
                 Py_DECREF(ind);
                 goto fail;
             }
-            offset = PyArray_PyIntAsInt(off);
+            long offset = PyArray_PyIntAsInt(off);
```

Comment:
```suggestion
            int offset = PyArray_PyIntAsInt(off);
```

Reply:
Would rather not fix that type of thing in this patch - `offset` was declared `long` before I moved the declaration, so I'd like this commit to not change its type.

Reply:
Fine, I had missed that it was `long`, because I remembered that the top of the function everything was `int`.

Reply:
... which is exactly the type of mistake I'm trying to avoid making myself by moving the types down to where they're used :)